crud/common/rebalance.lua

```diff
@@ -0,0 +1,211 @@
+local fiber = require('fiber')
+local log = require('log')
+local vshard_consts = require('vshard.consts')
+local utils = require('crud.common.utils')
+
+local has_metrics_module, metrics = pcall(require, 'metrics')
+
+local SETTINGS_SPACE_NAME = '_crud_settings'
+local SAFE_MOD_ENABLE_EVENT = '_crud.safe_mode_enable'
+
+local M = {
+    safe_mode = false,
+    safe_mode_enable_hooks = {},
+    safe_mode_disable_hooks = {},
+    _router_cache_last_clear_ts = fiber.time()
+}
+
+local function create_space()
+    local settings_space = box.schema.space.create(SETTINGS_SPACE_NAME, {
+        engine = 'memtx',
+        format = {
+            { name = 'key', type = 'string' },
+            { name = 'value', type = 'any' },
+        },
+        if_not_exists = true,
+    })
+    settings_space:create_index('primary', { parts = { 'key' }, if_not_exists = true })
+end
+
+local function safe_mode_trigger(_, new, space, op)
+    if space ~= '_bucket' then
+        return
+    end
+    if (op == 'INSERT' and new.status == vshard_consts.BUCKET.RECEIVING) or
+            (op == 'REPLACE' and new.status == vshard_consts.BUCKET.SENDING) then
+        box.broadcast(SAFE_MOD_ENABLE_EVENT, true)
+    end
+end
+
+local function register_enable_hook(func)
+    M.safe_mode_enable_hooks[func] = true
```

Comment:
Why do we need that variable and why cannot we just set `on_replace` trigger for `_bucket` on enable and remove it on disable?